### PR TITLE
Add remote kill switch for unsupported-OS native messaging

### PR DESCRIPTION
**Pull request by diegoreymendez**
Task/Issue URL: https://app.asana.com/1/137249556945/project/1206580121312550/task/1215328451378196?focus=true

### Description

Kill switch so we can stop the "os unsupported" messaging selectively.  This would allow us to hotfix the last Big Sur version and allow users to update.

### Testing Steps

1. Run a Debug build on macOS.
2. Open the internal feature flag overrides menu.
3. Enable osSupportForceUnsupportedMessage.
4. Open Preferences › About.
   - The unsupported-OS warning box is shown.
5. Back in the feature flag overrides menu, disable osSupportWarningEnabled.
6. Reopen Preferences › About.
   - The unsupported-OS warning box is no longer shown.

### Impact and Risks

Low. The flag defaults to enabled, so there is no behavior change unless it is explicitly disabled. The change gates only a warning UI — it cannot affect user data, privacy, or core functionality.

#### What could go wrong?

If the subfeature were disabled in config unintentionally, Big Sur users would stop seeing the end-of-support warning. This is mitigated by the default-enabled resolution (a missing subfeature resolves to on) and by the flag being scoped to the warning UI only — the OS support determination, the daily OS pixel, and RMF targeting are unaffected.

### Quality Considerations

The main edge case is the flag being absent from privacy config, which resolves to enabled; this is covered by unit tests alongside the kill-off and precedence cases. There is no performance impact (a single feature-flag read on an existing path) and no privacy/security surface, as only a UI warning is gated.

### Notes to Reviewer

The gate is a single guard at the SupportedOSChecker.unsupportedMinVersion chokepoint that every messaging surface consults, ordered before the existing debug force flag so the kill switch always wins. To operate it remotely, set macOSBrowserConfig.osSupportWarning to disabled.

---
###### Internal references:
[Definition of Done](https://app.asana.com/0/1202500774821704/1207634633537039/f) | [Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552) | [Tech Design Template](https://app.asana.com/0/59792373528535/184709971311943)

---

> [!NOTE]
> **Low Risk**
> Only gates warning UI via an existing chokepoint; default-on preserves current behavior unless privacy config disables the subfeature.
> 
> **Overview**
> Adds a remote **`osSupportWarning`** flag (privacy config `macOSBrowserConfig.osSupportWarning`, default **on**) wired through macOS `FeatureFlag` and `MacOSBrowserConfigSubfeature`.
> 
> **`SupportedOSChecker.unsupportedMinVersion`** now returns `nil` when the flag is off, which hides launch alerts, About/Feedback info boxes, and any other surface that keys off `showsSupportWarning`. That guard runs **before** the debug **`osSupportForceUnsupportedMessage`** override so the kill switch always wins.
> 
> Unit tests enable `.osSupportWarning` for existing cases and add coverage for kill-switch-off behavior (including ignoring the force flag).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 6e3075b21e9646c3e74dc816063e52bc5bdf8eec. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: macOS/DuckDuckGo/Common/OSVersion/SupportedOsChecker.swift [hunk 1]
```diff
@@ -200,6 +200,12 @@ extension SupportedOSChecker: SupportedOSChecking {
 
     var unsupportedMinVersion: String? {
 
+        // Remote kill switch: when disabled via privacy config, suppress all native OS-support
+        // messaging regardless of OS version or the debug force flag below.
+        guard featureFlagger.isFeatureOn(.osSupportWarning) else {
```
Comment (diegoreymendez) [inline macOS/DuckDuckGo/Common/OSVersion/SupportedOsChecker.swift:205]: Heart of the kill switch.

**Review by jleandroperez — APPROVED** [review]
:shipit:

File: macOS/LocalPackages/FeatureFlags/Sources/FeatureFlags/FeatureFlag.swift [hunk 2]
```diff
@@ -128,6 +128,10 @@ public enum FeatureFlag: String, CaseIterable {
     /// https://app.asana.com/1/137249556945/project/1211834678943996/task/1211866476152134
     case osSupportForceUnsupportedMessage
 
+    /// Remote kill switch for native unsupported-OS messaging. Enabled by default; disable via
+    /// privacy config (`macOSBrowserConfig.osSupportWarning`) to suppress the messaging.
+    case osSupportWarning
```
[the remark this answers is not in the document]
Comment (jleandroperez) [reply macOS/LocalPackages/FeatureFlags/Sources/FeatureFlags/FeatureFlag.swift:134]: I think this one is missing!

[the remark this answers is not in the document]
Comment (diegoreymendez) [reply macOS/LocalPackages/FeatureFlags/Sources/FeatureFlags/FeatureFlag.swift:134]: Resolved in https://github.com/duckduckgo/apple-browsers/pull/5229/commits/6e3075b21e9646c3e74dc816063e52bc5bdf8eec
